### ISSUE TITLE
[Windows] Attempted to send a key down event when no keys are in keysPressed.

### ISSUE THREAD

**Issue by Craftplacer** [flutter/flutter#94510]
## Steps to Reproduce

1. Type in a character using the numpad (further reference: https://en.wikipedia.org/wiki/Alt_code)

**Expected results:** Corresponding character appears
**Actual results:** Flutter crashes

<details>
  <summary>Logs</summary>

```
[  +49 ms] Building Windows application... (completed in 71.4s)
[ +888 ms] Observatory URL on device: http://127.0.0.1:19587/xrtS4_FUOYs=/
[  +23 ms] Caching compiled dill
[ +109 ms] Connecting to service protocol: http://127.0.0.1:19587/xrtS4_FUOYs=/
[ +321 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:19587/xrtS4_FUOYs=/.
[OMITTED]
[  +46 ms] DDS is listening at http://127.0.0.1:19590/34xjsOUxWck=/.
[ +162 ms] Successfully connected to service protocol: http://127.0.0.1:19587/xrtS4_FUOYs=/
[  +41 ms] DevFS: Creating new filesystem on the device (null)
[  +42 ms] DevFS: Created new filesystem on the device ([OMITTED])
[  +10 ms] Updating assets
[ +522 ms] Syncing files to device Windows...
[  +80 ms] <- reset
[   +3 ms] Compiling dart to kernel with 0 updated files
[  +48 ms] <- recompile [OMITTED]/.dart_tool/flutter_build/generated_main.dart 5fd93401-e149-44fe-82b4-62b961b9e0c6
[  +77 ms] <- 5fd93401-e149-44fe-82b4-62b961b9e0c6
[OMITTED]
[  +32 ms] Updating files.
[   +2 ms] DevFS: Sync finished
[  +62 ms] Syncing files to device Windows... (completed in 945ms)
[  +21 ms] Synced 0.0MB.
[  +20 ms] <- accept
[ +180 ms] Connected to _flutterView/0x21a35bed6d0.
[  +86 ms] Flutter run key commands.
[  +11 ms] r Hot reload. 
[   +6 ms] R Hot restart.
[   +6 ms] h List all available interactive commands.
[   +2 ms] d Detach (terminate "flutter run" but leave application running).
[   +7 ms] c Clear the screen
[  +25 ms] q Quit (terminate the application on the device).
[   +3 ms]  Running with sound null safety 
[  +13 ms] An Observatory debugger and profiler on Windows is available at: http://127.0.0.1:19590/34xjsOUxWck=/
[  +22 ms] Activating Dart DevTools...
[OMITTED]
[+4217 ms] Unable to parse JSON message:
[   +6 ms] The document is empty.
[  +32 ms] 
           ══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
           The following assertion was thrown during a platform message callback:
           Attempted to send a key down event when no keys are in keysPressed. This state can occur if the key
           event being sent doesn't properly set its modifier flags. This was the event:
           RawKeyDownEvent#30887(logicalKey: LogicalKeyboardKey#00237(keyId: "0x200000237", keyLabel: "Numpad
           7", debugName: "Numpad 7"), physicalKey: PhysicalKeyboardKey#700e2(usbHidUsage: "0x000700e2",
           debugName: "Alt Left")) and its data: RawKeyEventDataWindows#3635f(keyCode: 103, scanCode: 56,
           characterCodePoint: 8226, modifiers: 0)
           'package:flutter/src/services/raw_keyboard.dart':
           Failed assertion: line 668 pos 7: 'event is! RawKeyDownEvent || _keysPressed.isNotEmpty'

           Either the assertion indicates an error in the framework itself, or we should provide substantially
           more information in this error message to help you determine and fix the underlying cause.
           In either case, please report this assertion by filing a bug on GitHub:
             https://github.com/flutter/flutter/issues/new?template=2_bug.md

           When the exception was thrown, this was the stack:
           #2      RawKeyboard.handleRawKeyEvent (package:flutter/src/services/raw_keyboard.dart:668:7)
           #3      KeyEventManager.handleRawKeyMessage (package:flutter/src/services/hardware_keyboard.dart:818:33)
           #4      BasicMessageChannel.setMessageHandler.<anonymous closure> (package:flutter/src/services/platform_channel.dart:73:49)
           #5      BasicMessageChannel.setMessageHandler.<anonymous closure> (package:flutter/src/services/platform_channel.dart:72:47)
           #6      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure> (package:flutter/src/services/binding.dart:379:35)
           #7      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure> (package:flutter/src/services/binding.dart:376:46)
           #8      _invoke2.<anonymous closure> (dart:ui/hooks.dart:205:15)
           #12     _invoke2 (dart:ui/hooks.dart:204:10)
           #13     _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:42:5)
           #14     _Channel.push (dart:ui/channel_buffers.dart:132:31)
           #15     ChannelBuffers.push (dart:ui/channel_buffers.dart:329:17)
           #16     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:544:22)
           #17     _dispatchPlatformMessage (dart:ui/hooks.dart:92:31)
           (elided 5 frames from class _AssertionError and dart:async)
           ════════════════════════════════════════════════════════════════════════════════════════════════════
[+2554 ms] Service protocol connection closed.
[  +63 ms] Lost connection to device.
[  +12 ms] DevFS: Deleting filesystem on the device ([OMITTED])
[ +138 ms] Activating Dart DevTools... (completed in 9.9s)
[ +138 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed
[  +37 ms] "flutter run" took 89,757ms.
[ +242 ms] ensureAnalyticsSent: 222ms
[  +35 ms] Running shutdown hooks
[   +2 ms] Shutdown hooks complete
[  +17 ms] exiting with code 0
```

```
No issues found! (ran in 5.7s)
```

```
[√] Flutter (Channel stable, 2.5.3, on Microsoft Windows [Version 10.0.19043.1348], locale en-DE)
    • Flutter version 2.5.3 at C:\SDKs\Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (7 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\SDKs\Android
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • CHROME_EXECUTABLE = C:\Users\Craftplacer\AppData\Local\Chromium\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.0)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31605.320
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[☠] IntelliJ IDEA Community Edition (the doctor check crashed)
    X Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know about this issue at
      https://github.com/flutter/flutter/issues.
    [Omitted]


[√] VS Code, 64-bit edition (version 1.59.0)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.27.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1348]
    • Chrome (web)      • chrome  • web-javascript • unknown
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 96.0.1054.34

! Doctor found issues in 1 category.
```

</details>

https://user-images.githubusercontent.com/22963120/144330269-db78d2f8-33be-4813-a4d9-af879bf7586c.mp4

**Comment by maheshmnj** [comment]
Hi @Craftplacer, Thanks for filing the issue, I ran the newly created counter app on windows as a Desktop app and tried pressing Numpad keys and also by holding the alt key but I could not see any error in the console. Could you please specify the steps to reproduce the issue?
 

<details>
<summary>flutter doctor -v</summary>

```
[√] Flutter (Channel stable, 2.5.3, on Microsoft Windows [Version 10.0.19042.1348], locale en-US)
    • Flutter version 2.5.3 at C:\flutter_sdk\stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (7 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java      
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.61.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.28.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1348]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 96.0.4664.45
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 95.0.1020.44

• No issues found!
```
</details>

Thank you.

**Comment by Craftplacer** [comment]
There are no additional steps "necessary", Flutter crashes as soon as Windows tries to input a character (for example by pressing Numpad 7 while holding Alt and having Num Lock enabled)

**Comment by maheshmnj** [comment]
> Flutter crashes as soon as Windows tries to input a character (for example by pressing Numpad 7 while holding Alt and having Num Lock enabled)

Thanks for the info @Craftplacer, enabling Numlock is crucial info. I was able to reproduce the issue on stable 2.5.3, but on the master channel this is not reproducible, It seems like this is fixed as of `master  2.6.0-12.0.pre.885`. You may either wait until the fix lands on the stable or switch to the master channel using 
```
flutter channel master
flutter upgrade
``` 

<details>
<summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 2.6.0-12.0.pre.885, on Microsoft Windows [Version 10.0.19042.1348], locale    
    en-US)
    • Flutter version 2.6.0-12.0.pre.885 at C:\flutter_sdk\master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 16ec56ee0f (in the future), 2021-12-02 02:49:05 -0500
    • Engine revision 888f4c0fc6
    • Dart version 2.16.0 (build 2.16.0-56.0.dev)
    • DevTools version 2.8.0

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.61.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.28.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1348]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 96.0.4664.45
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 95.0.1020.44

• No issues found!
```
</details>

 Closing as fixed on the master channel. If you still encounter the issue, feel free to write back and I will reopen it.
 
 Thank you
